### DISCUSSION TITLE
How to perform a backup using Flatpak Vorta through CLI

### DISCUSSION THREAD

**Discussion by irfanhakim-as** [borgbase/vorta#1840] [FAQ]
## Overview

The Vorta application used for this question is the Flatpak version in particular.

I've used Vorta (rather than Borg) as my desktop backup solution and I intend to continue doing so, but I'm wondering if it's possible for me to continue using it (even if indirectly i.e. through Borg) to perform backups when I only have access to a non-graphical session (i.e. through SSH).

## Use case

My desktop in this case is a Steam Deck. I installed Vorta as a Flatpak and I use it to backup my user files to my backup repository, this gets done periodically automatically but only when I'm in desktop mode. Most of the time I'd be in Gaming mode so my Vorta backups won't run when that's the case. In such situations, my idea is to SSH into my Steam Deck from my primary computer, and run the exact same command Vorta uses to perform the backup. For example:

```sh
/app/bin/borg create --list --progress --info --log-json --json --filter=AM -C lz4 --exclude-from /tmp/tmp7ajkp9fx --exclude-if-present .nobackup /mnt/backup/foo::foo-2023-11-03-171248 /home/deck
```

I'm expecting that running the exact same command Vorta uses should work, though there are 2 things that come to mind:

- `/app/bin/borg` obviously wouldn't work and does not exist since Vorta was installed as a Flatpak and Borg isn't actually installed in the system.
- Not sure if this is the "ideal" way of doing this.

## Question

So now I'm hoping to get opinions/solutions from Vorta experts on what I can do to essentially do the same thing I do on Vorta but via the terminal, having only Vorta installed on the system as a Flatpak? Thank you.

**Comment by m3nu** [comment]
To run Vorta, you need a graphical environment/X server. For a headless machine, it would be Borg only.

**Reply by irfanhakim-as** [reply]
Would it be possible to run the command I've included then to perform the backup using Borg bundled with the Vorta Flatpak? I assume the created archives would still appear in the Vorta UI afterwards doing it that way?

Though I'd need to figure out how to use _that_ Borg. Is the "recommended" way of doing so is to replace the `/app/bin/borg` path used by Vorta with maybe, `/var/lib/flatpak/app/com.borgbase.Vorta/current/active/files/bin/borg`?
